src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -82,7 +82,10 @@ public function __construct(
         $this->securityChecker = $securityChecker;
     }
 
-    public function getFilters()
+    /**
+     * @return TwigFilter[]
+     */
```

Comment:
Please contribute this to 3.x, and anything else that is backward-compatible

Reply:
Cant we ask for a PR on 3.x without blocking this one ?

The last time we did this, it was with a PR of @jordisala1991 and @core23.
It delayed a lot the Pr on master because of the conflict.

When our goal is to release Sonata 4, Im dont see as needed to report everything to 3.x (especially when new contributors are helping us) but I'm maybe the only one to think this way.

Reply:
TLDR: to me a Pr on 3.x is a bonus point, not a blocking point

Reply:
To me it's blocking: didn't you complain there were to many merge up conflicts the other day? Contributing things to `master` that could be contributed to `3.x` is a recipe for conflicts. To me the workflow should be
- contribute BC things to `3.x`
- wait for the merge up (the contributor can ask us to speed that up if they are eager to contribute)
- contribute non-BC things to `master`.

Reply:
> To me it's blocking: didn't you complain there were to many merge up conflicts the other day? Contributing things to `master` that could be contributed to `3.x` is a recipe for conflicts. To me the workflow should be
> 
> - contribute BC things to `3.x`
> 
> - wait for the merge up (the contributor can ask us to speed that up if they are eager to contribute)
> 
> - contribute non-BC things to `master`.

That workflow delay a lot the work on sonata 4 IMO

Reply:
I did complain about conflict from merge up. But Im not sure this workflow will help.

The more we're doing PR on 3.x the more we will have conflict because of deprecated method, phpdoc already updated, ...

When core23 followed your workflow, I ended up with lot of conflict during the merge and jordi had lot of conflict during the rebase. 

And I agree with jordi that it delays Sonata 4. Which urge, because we're still not Symfony 5 compatible.

I would prefer to not delay contributor contribution on Master because we're not a lot working on this...

Reply:
> The more we're doing PR on 3.x the more we will have conflict because of deprecated method, phpdoc already updated, ...

Conflicts happen because branches diverge, this makes no sense to me. While I'd agree that you cannot have conflicts if you have never anything to merge up, if anyone decides to contribute some of the changes in this PR, but slightly different on 3.x, you will have a conflict. So in the long run, not doing most of the work on 3.x results in more conflicts. If you are not convinced, please try merging up doctrine/orm 2.8.x into master, you will see what I mean.

> When core23 followed your workflow, I ended up with lot of conflict during the merge and jordi had lot of conflict during the rebase.

Because that workflow was not followed perfectly before, maybe? That might also happen because 3.0.0 is very old.

> And I agree with jordi that it delays Sonata 4. Which urge, because we're still not Symfony 5 compatible.

Anyway, while the choices you are making are harmful in the long run IMO, your argument that there won't be a long run because 4.0.0 is happening soon is valid, and you are more involved than I in fulfilling that goal. I will trust your judgement and dismiss my review.

Once 4.0.0 is out though, let's try again following a workflow that strongly discourages contributing anything backwards compatible to master, otherwise we will end in the same painful situation.

Reply:
> Once 4.0.0 is out though, let's try again following a workflow that strongly discourages contributing anything backwards compatible to master, otherwise we will end in the same painful situation.

Im 100% with you on this.
Imho the first thing to do will be to release major more often.